src/Traits/GetDependenciesTrait.php

```diff
@@ -22,7 +22,7 @@ protected function transformDependency(ReflectionParameter $parameter)
     {
         $type = $parameter->getType();
 
-        if (empty($type)) {
+        if (empty($type) || !$type instanceof \ReflectionNamedType) {
```

Comment:
if this change fixes some bug - please make it as a separate PR and cover it by autotests